packages/visual-editor/src/vite-plugin/plugin.ts

```diff
@@ -4,28 +4,137 @@ import { Plugin } from "vite";
 import mainTemplate from "./templates/main.tsx?raw";
 import editTemplate from "./templates/edit.tsx?raw";
 import directoryTemplate from "./templates/directory.tsx?raw";
+import { ComponentField, ComponentFieldMappings } from "@yext/visual-editor";
+import { defaultLayoutData } from "./defaultLayoutData.ts";
+
+type TemplateDefinition = {
+  name: string;
+  description: string;
+  content: string;
+  layoutRequired: boolean;
+  includeInManifest: boolean;
+  exampleSiteUrl?: string;
+  defaultLayoutData?: string;
+  componentFields?: Array<ComponentField>;
+};
 
 // files to generate
-const virtualFiles: Record<string, string> = {
-  "src/templates/main.tsx": mainTemplate,
-  "src/templates/edit.tsx": editTemplate,
-  "src/templates/directory.tsx": directoryTemplate,
+const templateDefinitions: Record<string, TemplateDefinition> = {
+  "src/templates/main.tsx": {
+    name: "main",
+    description:
+      "Use this template to generate pages for each of your Locations.",
+    content: mainTemplate,
+    layoutRequired: true,
+    includeInManifest: true,
+    exampleSiteUrl: "",
+    defaultLayoutData: defaultLayoutData.main,
+    componentFields: [
+      ComponentFieldMappings.HeroSection,
+      ComponentFieldMappings.PromoSection,
+      ComponentFieldMappings.ProductsSection,
+      ComponentFieldMappings.EventsSection,
+      ComponentFieldMappings.FAQSection,
+      ComponentFieldMappings.TestimonialsSection,
+      ComponentFieldMappings.InsightsSection,
+      ComponentFieldMappings.TeamSection,
+    ],
+  },
+  "src/templates/edit.tsx": {
+    name: "edit",
+    description: "The template used to generate the layout and theme editor.",
+    content: editTemplate,
+    layoutRequired: false,
+    includeInManifest: false,
+  },
+  "src/templates/directory.tsx": {
+    name: "directory",
```

Comment:
We don't use the component fields here because Spruce handles this one specifically on their end because of the dynamic nature of the fields? Maybe comment that.